### PR TITLE
Feat/raw response type directive

### PR DESCRIPTION
**Pull request by zth**

### CODE REVIEW

File: packages/reason-relay/language-plugin/src/generator/Printer.re [hunk 1]
```diff
@@ -174,19 +174,16 @@ and printObject =
          }
        )
     |> Array.iteri((index, p) => {
```
Comment (tsnobip) [inline packages/reason-relay/language-plugin/src/generator/Printer.re:176]: not that you really have perf issues here, but Belt versions of array functions are usually more optimized.

File: packages/reason-relay/src/ReasonRelay.re [hunk 2]
```diff
@@ -1142,34 +1146,34 @@ type optimisticUpdaterFn = RecordSourceSelectorProxy.t => unit;
...
   onUnsubscribe: option(unit => unit),
-  optimisticResponse: option('response),
+  optimisticResponse: option('rawResponse),
   optimisticUpdater: option(optimisticUpdaterFn),
   updater: option(updaterFn('response)),
   variables: 'variables,
 };
 
-type _useMutationConfig('response, 'variables) = {
+type _useMutationConfig('response, 'rawResponse, 'variables) = {
```
Comment (tsnobip) [inline packages/reason-relay/src/ReasonRelay.re:1159]: not really new, but we could use a clearer naming convention for types used only for bindings, like `myTypeRaw` or `myTypeJs`

File: packages/reason-relay/src/ReasonRelay.re [hunk 3]
```diff
@@ -1178,14 +1182,20 @@ exception Mutation_failed(array(mutationError));
 
 [@bs.module "relay-runtime"]
 external commitMutation_:
```
Comment (tsnobip) [inline packages/reason-relay/src/ReasonRelay.re:1184]: same as for types, we could follow the same naming convention or just use shadowing to be sure that the raw binding is not used elsewhere.

File: packages/reason-relay/language-plugin/src/__tests__/languagePlugin-tests.ts [hunk 4]
```diff
@@ -64,17 +64,18 @@ function generate(text: string, options?: any, extraDefs: string = "") {
...
+  return ctx
     .documents()
     .map(
       (doc: any) =>
         `// ${doc.name}.graphql\n${printCode(
           RelayReasonGenerator.generate(extendedSchema, doc, {
-            customScalars: {},
+            normalizationIR: ctx.get(doc.name),
             optionalInputFields: [],
-            existingFragmentNames: new Set([]),
```
[the remark this answers is not in the document]
Comment (zth) [reply packages/reason-relay/language-plugin/src/__tests__/languagePlugin-tests.ts:77]: It originally came from some internal Relay code, and I'm tbh not sure what it does 😄 So I removed it and things seemed to continue working.

File: packages/reason-relay/language-plugin/src/generator/Printer.re (hunk 1 above)
Comment (zth) [reply packages/reason-relay/language-plugin/src/generator/Printer.re:176]: Ah nice, I'll change that! It's not totally unlikely that I'll end up porting this to native Reason at some point when the new Relay compiler ships. But let's cross that bridge when we get to it.

File: packages/reason-relay/src/ReasonRelay.re (hunk 2 above)
Comment (zth) [reply packages/reason-relay/src/ReasonRelay.re:1159]: Yup, agree! Probably best as a separate PR though.

File: packages/reason-relay/src/ReasonRelay.re (hunk 3 above)
Comment (zth) [reply packages/reason-relay/src/ReasonRelay.re:1184]: Agreed! And separate-pr-material 🤪

Comment (tsnobip) [reply packages/reason-relay/src/ReasonRelay.re:1184]: haha yeah, a new cleanup PR for `ReasonRelay.re` in perspective :)

File: packages/reason-relay/language-plugin/src/__tests__/languagePlugin-tests.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tsnobip) [reply packages/reason-relay/language-plugin/src/__tests__/languagePlugin-tests.ts:77]: haha whatever works, it's always good to remove unused code :)

**Pull request merged**